flaml/autogen/agentchat/responsive_agent.py

```diff
@@ -110,24 +133,12 @@ def __init__(
         self._default_auto_reply = default_auto_reply
         self._class_specific_reply = []
         self.reply_at_receive = defaultdict(bool)
-        self.register_auto_reply(Agent, self._generate_oai_reply)
-        self.register_auto_reply(Agent, self._generate_code_execution_reply)
-        self.register_auto_reply(Agent, self._generate_function_call_reply)
-        self.register_auto_reply(Agent, self._check_termination_and_human_reply)
-
-    def register_auto_reply(self, class_type, reply_func: Callable, position: int = 0):
-        """Register a class-specific reply function.
 
-        The class-specific reply function will be called when the sender is an instance of the class_type.
-        The function registered later will be checked earlier by default.
-        To change the order, set the position to a positive integer.
-
-        Args:
-            class_type (Class): the class type.
-            reply_func (Callable): the reply function.
-            position (int): the position of the reply function in the reply function list.
-        """
-        self._class_specific_reply.insert(position, (class_type, reply_func))
+        # Handle class-specific reply defined in the "register_auto_reply" decorator.
+        for cls in reversed(type(self).__mro__):  # loop all supers
```

Comment:
How about removing this line to avoid repeated registration?

Reply:
Add a list to avoid repeated registration.